### PR TITLE
[r] Config and context support for `SOMACollection`'s `add_new_*()` methods

### PR DESCRIPTION
**Pull request by aaronwolen**
**Issue and/or context:** #1447

**Changes:** Previously, the `add_new_*()` methods of `SOMACollection` did not pass through the parent context or platform config. This has been fixed. The  `add_new_*()` methods also now provide a `platform_config` argument to override the parent's config options.

A [new section](https://github.com/single-cell-data/TileDB-SOMA/pull/1448/files#diff-963debf4c7662e18cfe8041bf9c8ff6d4c32a29e2bcc9120aeee8e90a5b6b46d) explaining this behavior has been added to the relevant classes.

- Code from the test helper `create_and_populate_soma_dataframe()` that created the test arrow table was moved to a separate helper function: `create_test_arrow_table()`. This new function is utilized in the new unit tests for `SOMACollection`'s `add_new_*()` methods.
- Updated SOMACollection's description to linkify referenced classes
- Renamed `TileDBObject`'s private `tiledb_platform_config` field was renamed to `. tiledb_platform_config` for consistency with `.tiledbsoma_ctx`

### CODE REVIEW

File: apis/r/R/SOMACollection.R [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-#' @description Contains a key-value mapping where the keys are string names
-#' and the values are any SOMA-defined foundational or composed type, including
-#' `SOMACollection`, `SOMADataFrame`, `SOMADenseNDArray`, `SOMASparseNDArray`
-#' or `SOMAExperiment`.  (lifecycle: experimental)
+#' @description Contains a key-value mapping where the keys are string names and
+#' the values are any SOMA-defined foundational or composed type, including
+#' [`SOMACollection`], [`SOMADataFrame`], [`SOMADenseNDArray`],
+#' [`SOMASparseNDArray`], or [`SOMAExperiment`].  (lifecycle: experimental)
+#'
+#' @templateVar class SOMACollection @template section-add-object-to-collection
```
Comment (aaronwolen) [inline apis/r/R/SOMACollection.R:8]: Linkified class references

File: apis/r/DESCRIPTION [hunk 2]
```diff
@@ -4,7 +4,7 @@ Title: TileDB SOMA
 Description: Interface for working with 'TileDB'-based Stack of Matrices,
     Annotated ('SOMA'): an open data model for representing annotated matrices,
     like those commonly used for single cell data analysis.
-Version: 0.0.0.9025
+Version: 0.0.0.9026
```
Comment (eddelbuettel) [inline apis/r/DESCRIPTION:7]: :+1:

File: apis/r/man/SOMACollection.Rd [hunk 3]
```diff
@@ -9,8 +9,21 @@ and the values are any SOMA-defined foundational or composed type, including
 \code{SOMACollection}, \code{SOMADataFrame}, \code{SOMADenseNDArray}, \code{SOMASparseNDArray}
 or \code{SOMAExperiment}.  (lifecycle: experimental)
 }
+\section{Adding new objects to a collection}{
+
+
+The \code{\link{SOMACollection}} class provides a number of type-specific methods for
+adding new object to the collection, such as \code{add_new_sparse_ndarray()} and
```
Comment (eddelbuettel) [inline apis/r/man/SOMACollection.Rd:16]: Micro-edit:  "adding _a_ new object" (or else plural "adding new _objects_")

File: apis/r/man/roxygen/templates/section-add-object-to-collection.R [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#' @section Adding new objects to a collection:
+#'
+#' The [`<%= class %>`] class provides a number of type-specific methods for
+#' adding new object to the collection, such as `add_new_sparse_ndarray()` and
```
Comment (eddelbuettel) [inline apis/r/man/roxygen/templates/section-add-object-to-collection.R:4]: As above

**Review by eddelbuettel — APPROVED** [review]
Looks good; I added a few minor editorial comments.  Of course CI will have the last word :)

**Comment by aaronwolen** [comment]
Thanks, @eddelbuettel!

**Pull request merged**